refactor(translations): NASS-1228:  Mobile + Web translations logic consistency

mlie788 opened the pull request:
### Changes

The main comment on the ticket was that replacements for translations in mobile were handled in the component, but it seems like it has already been moved to the mobile translation, so what I have done is refactor the replacements logic with the replacementsConfig stuff from desktop so that setting uppercase is handled there rather than the component.

### Deploys

- [x] **Deploy to Tamanu Internal** <!-- #deploy -->

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

<!-- Thank you! -->

mlie788 committed:
refactor: updated TranslatedText and Translation context for mobile to be consistent in replacement config

tcodling reviewed (changes requested):
great job on this. This has actually changed quite a lot since i worked on it so its quite complicated 😆 Main thing is I would try to also include that `lowercase` logic in the apply casing that you copied from desktop just to keep things consistent.

mlie788 committed:
refactor: applied changes suggested from code review

Adding logic for lowercase, cleaning up commented code, applied use of isValidElement for string replacement

tcodling reviewed (approved):
LGTM! will take over this card and post testing notes
